c/prima.c

```diff
@@ -21,50 +56,127 @@ int lincoa_c(prima_obj calfun, const void *data, const int n, double x[], double
              const double xl[], const double xu[],
              int *nf, const double rhobeg, const double rhoend, const double ftarget, const int maxfun, const int npt, const int iprint, int *info);
 
+int prima_check_options(prima_options * opt, int n)
+{
+  if (!opt)
+    return PRIMA_NULL_OPTIONS;
+  if (!opt->xl)
+  {
+    opt->xl = (double*)malloc(n * sizeof(double));
```

Comment:
The value returned by `malloc` should be tested. `NULL` indicates a failure (insufficient memory).

Reply:
This field is always allocated, howver it is not needed for `uobyqa` and `newuoa`. For the other Powell's algorithms, I do not know the behavior of the Fortran code: are the bounds mandatory or can they be omitted?

The same remarks apply for the `xu` and `nlconstr0` fields.

Reply:
yes, lets add a flag to initialize bounds